pkg/storage/tsdb/config.go

```diff
@@ -302,6 +302,8 @@ type BucketStoreConfig struct {
 
 	// Controls what to do when MaxConcurrent is exceeded: fail immediately or wait for a slot to run.
 	MaxConcurrentRejectOverLimit bool `yaml:"max_concurrent_reject_over_limit" category:"experimental"`
+	BatchedSeriesLoading         bool `yaml:"batched_series_loading" category:"experimental"`
```

Comment:
Stylistic: I'd rather disable the feature by setting the batch size to 0 instead of having two flags to control it.